### PR TITLE
chore: Prepare 3.1.0 release.

### PR DESCRIPTION
**Pull request by mikeproeng37**
## Summary
- Bump version
- Add changelog changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+    * Feature access for users.
+    * Feature variable value for users.
+    * Added APIs to be able to conveniently add Decision notification handler (addDecisionNotificationHandler) and Track notification handler (addTrackNotificationHandler).
+
+## Bug Fixes:
+* Feature variable APIs return default variable value when featureEnabled property is false. (#390)
+
+## Deprecated
+* Activate notification listener is deprecated as of this release. Recommendation is to use the new Decision notification listener. Activate notification listener will be removed in the next major release.
+* addActivateNotificationListener, addTrackNotificationListener and addNotificationListener APIs on NotificationCenter.
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: I don't think we refactored code in this SDK. Please take a look.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+
+## New Features:
+* Introduced Decision notification listener to be able to record:
+    * Variation assignments for users activated in an experiment.
+    * Feature access for users.
+    * Feature variable value for users.
+    * Added APIs to be able to conveniently add Decision notification handler (addDecisionNotificationHandler) and Track notification handler (addTrackNotificationHandler).
+
+## Bug Fixes:
+* Feature variable APIs return default variable value when featureEnabled property is false. (#390)
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: Link to PR missing.

**Review by aliabbasrizvi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mikeng13) [reply CHANGELOG.md]: We added a deprecation notice in the header: https://github.com/optimizely/objective-c-sdk/blob/master/OptimizelySDKCore/OptimizelySDKCore/OPTLYNotificationCenter.h#L101

Comment (mikeproeng37) [reply CHANGELOG.md]: You are right, we only added a deprecation notice in the `addActivateNotificationListener` method: https://github.com/optimizely/objective-c-sdk/blob/master/OptimizelySDKCore/OptimizelySDKCore/OPTLYNotificationCenter.h#L101

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+    * Feature access for users.
+    * Feature variable value for users.
+    * Added APIs to be able to conveniently add Decision notification handler (addDecisionNotificationHandler) and Track notification handler (addTrackNotificationHandler).
+
+## Bug Fixes:
+* Feature variable APIs return default variable value when featureEnabled property is false. ([#390](https://github.com/optimizely/objective-c-sdk/pull/390))
+
+## Deprecated
+* Activate notification listener is deprecated as of this release. Recommendation is to use the new Decision notification listener. Activate notification listener will be removed in the next major release.
+* addActivateNotificationListener API on NotificationCenter.
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: This point is redundant as it is captured in the previous line.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
 
+## 3.1.0
+May 15th, 2019
+
+## New Features:
+* Introduced Decision notification listener to be able to record:
+    * Variation assignments for users activated in an experiment.
+    * Feature access for users.
+    * Feature variable value for users.
+    * Added APIs to be able to conveniently add Decision notification handler (addDecisionNotificationHandler) and Track notification handler (addTrackNotificationHandler).
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: This point does not apply to this SDK.

**Review by aliabbasrizvi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jaeopt) [inline CHANGELOG.md]: What is the meaning of this 2nd bullet of "Deprecated"? Not clear to me.

**Review by jaeopt — COMMENTED** [review]
see comments. 
Also we usually change version number in Xcode project in sync with podspec version. I just want to make sure they are all updated correctly.

**Pull request closed without merging**